src/OSPSuite.Core/Domain/Services/ParameterValuesCreator.cs

```diff
@@ -72,22 +80,19 @@ public ParameterValuesCreator(IIdGenerator idGenerator, IEntityPathResolver enti
          return parameterValue;
       }
 
-      public IReadOnlyList<ParameterValue> CreateFrom(SpatialStructure spatialStructure, IReadOnlyList<MoleculeBuilder> molecules)
-      {
-         var parameterValues = new List<ParameterValue>();
-         spatialStructure.PhysicalContainers.Each(container => addParametersFrom(parameterValues, container, molecules));
-         return parameterValues;
-      }
+      public IReadOnlyList<ParameterValue> CreateFrom(SpatialStructure spatialStructure, IReadOnlyList<MoleculeBuilder> molecules) => 
+         spatialStructure.PhysicalContainers.SelectMany(container => createFrom(container, molecules, isLocalWithConstantFormula)).ToList();
 
-      private void addParametersFrom(List<ParameterValue> parameterValues, IContainer container, IReadOnlyList<MoleculeBuilder> molecules) => 
-         molecules.Each(x => addParametersFrom(parameterValues, container, x));
+      public IReadOnlyList<ParameterValue> CreateExpressionFrom(IContainer physicalContainer, IReadOnlyList<MoleculeBuilder> molecules) => 
+         physicalContainer.GetAllContainersAndSelf<IContainer>(x => x.Mode.Equals(ContainerMode.Physical)).SelectMany(container => createFrom(container, molecules, x => x.IsExpression())).ToList();
```

Comment:
using equals on an enum feels very weird to be honest. 
Aslo don't we have an extension method such as Is ?

Reply:
Not for ContainerMode it seems. But I can make it.